### ISSUE TITLE
%f macro handling is inconsistent with spaces

### ISSUE THREAD

**Issue by laktak** [ranger/ranger#1935]
#### Runtime Environment

- Operating system and version: macOS 10.15
- Terminal emulator and version:  iTerm 3
- Python version: 3.7.7
- Ranger version/commit: ranger-master v1.9.2-436-g3f8e7c14
- Locale: en

#### Current Behavior

```
map KEY shell COMMAND %f
map KEY console shell COMMAND %f
map KEY chain mark_files all=True val=False; shell COMMAND %f
```
If the selected file contains spaces, only the first mapping will work correctly.

For the second and third mappings the `COMMAND` does not receive the correct file argument.

#### Expected Behavior

`%f` should work the same in any case.

This affects other macros as well (like `%s`).

#### Possible Solutions

I can escape `%f` myself but I don't have a workaround when using `%s`

```
map KEY console shell COMMAND %s
```

**Comment by toonn** [comment]
This has been brought up in #1931.

The reason is that shell escaping *shouldn't* happen in some cases.
For example, `:touch %s` on a file named `file 1`, should *not* have escaping applied to the macro expansion because rather than touching `file 1` you'd be creating a file named `'file 1'`, *including* the apostrophes.

To avoid this problem when using console you should escape the macros `%%s`.

We'll add a flag in the future because people don't seem to understand when shell escaping is necessary and when it's not.

**Comment by laktak** [comment]
I looked for issues but didn't see the PR about this. I see that this is not as simple as I thought... but it solves my issue, so you can close this if you want.

I still think that escaping the macro is not ideal though. When I have a console command I wan't to see the file name and not a macro like `%f`.

**Comment by toonn** [comment]
Yeah, that's the same thing the author of that PR wants. You could work around it using `eval` but that's not very nice.

**Comment by mrgoodvin** [comment]
***toonn***, you're right. Indeed, everything is not as simple as it seemed to me. I thought for a long time but did not found a good solution. _will not have time for the next 4-5 months. I hope that until this time what this issue will still be relevant :). It seems to me that a good solution simply does not exist. As I see something like this:

Handling crazy file/directory names in %s on hotkeys in shell | console, like it — "file '1' "name "" "1'
```
"file '1" => normal
'file "1' => normal
"file "1' => crazy
"file "1" => crazy
```
To way realisation:

1. Mixed handling (canonical + dividing) — if any one is a crazy on checkup => dividing method for %s
   1. checkup of perhaps having crazy file name && and process of retention selected method
      1. => normal
	 `'file1' 'file2'`
	 `"file1" "file2" #*1`
	 => canonical view/handling

      2. => crazy => delim = `_`
	 => dividing view, console example. For one file perhaps will be actual current ranger version. The user chooses to use two crazy file names, he will need to switch to another option.
      	 `cfile 1 #'cfile 1'`
      	 `cfile__1 #'cfile_1'`
      	 `cfile___1 #'cfile__1'`
      	 `cfile___ 1 #'cfile__ 1'`
	 `#    view  #  cfile name`
	 `#2   __    #1 '_'`
	 `#3   ___   #2 '__'`
	 `#4   ____  #3 '___'`
	 `#n+1 _____ #n '____`
      	 `cfile1_cfile2       #'cfile1' 'cfile2'`
      	 `cfile 1_cfile 2     #'cfile 1' 'cfile 2'`
	 `cfile__1_cfile 2    #'cfile_1' 'cfile 2'`
      	 `cfile__1_cfile__ 2  #'cfile_1' 'cfile_ 2'`
      	 `cfile__1_cfile___ 2 #'cfile_1' 'cfile__ 2'`
	 `"file '1' "name "" "1'_"file '1' "name "" "2' #*2`
	 => dividing handling => `__` is replaced on `_` with saving a position of delimiter
	 => dividing handling (maybe use ranger method instead shell version) => "file\ \'1\'\ \"name\ \"\"\ \"1\' "file\ \'1\'\ \"name\ \"\"\ \"2\' #*3
   2. preferred... (hotkey of console for switching(change) of handling method normal/dividing[/shell] selected if available of cause; + indication of dividing-method; + modification console text line on switching *1,*2[,*3], maybe)

2. Divide-handling on all time (not preferred for normal human perception).
Crazy file names are created only by crazy people :).

**Comment by toonn** [comment]
My current fix for this would be to have `:console <cmd>` inherit the shell escaping behavior of `<cmd>`, so when you do `console shell %s`, the files in the selection would be properly shell escaped. If you do `console touch %f` there will be no shell escaping.
